src/main/java/Duke.java

```diff
@@ -1,10 +1,411 @@
+
+    public static void clearDatabase() {
+        try {
+            FileWriter fw = new FileWriter(FILEPATH);
+            fw.write("");
+            fw.close();
+        } catch (IOException e) {
+            System.out.println(LINES + NO_FILE_MESSAGE + LINES);
+        }
+    }
+
+    public static void updateDatabase() {
+        //update after mark done.
+        //clear file then append from list
+        clearDatabase();
+        for (Task task: tasks) {
+            String isDone = task.isDone? "1" : "0";
+            if (task instanceof Todo) {
+                appendDatabase("T", task.description, isDone);
+            } else if (task instanceof Deadline) {
+                Deadline deadline = (Deadline) task;
+                String input = deadline.description + "/by" + deadline.getDeadline();
+                appendDatabase("D", input , isDone);
+            } else if (task instanceof Event) {
+                Event event = (Event) task;
+                String input = event.description + "/at" + event.getDeadline();
+                appendDatabase("E", input , isDone);
+            }
+        }
+    }
+
+    public static void initList(Scanner s) throws JimException {
+        while (s.hasNext()) {
+            String task = s.nextLine();
+            String[] listedTask = task.split(" \\| ", 4);
+            switch (listedTask[0]) {
+            case "T":
+                tasks.add(new Todo(listedTask[2]));
+                if (listedTask[1].equals("1")) {
+                    tasks.get(tasks.size() - 1).markAsDone();
+                }
+                break;
+            case "D":
+                tasks.add(new Deadline(listedTask[2], listedTask[3]));
+                if (listedTask[1].equals("1")) {
+                    tasks.get(tasks.size() - 1).markAsDone();
+                }
+                break;
+            case "E":
+                tasks.add(new Event(listedTask[2], listedTask[3]));
+                if (listedTask[1].equals("1")) {
+                    tasks.get(tasks.size() - 1).markAsDone();
+                }
+                break;
+            default:
+                throw new JimException();
+            }
+        }
+        s.close();
+    }
+
+    //Initialises the list
+    public static void initJim() {
+        try {
+            tasks = new ArrayList<Task>();
+            File file = new File(FILEPATH);
+            File folder = new File(FOLDERPATH);
+            folderChecker(folder);
+            databaseChecker(file);
+            Scanner s = new Scanner(file);
+            initList(s);
+        } catch (JimException e) {
+            System.out.println(LINES + "File is corrupted at Line " + (tasks.size()+1) + "!" +
+                    System.lineSeparator() + LINES);
+        } catch (FileNotFoundException e) {
+            System.out.println(LINES + NO_FILE_MESSAGE + LINES);
+        }
+    }
+
+    public static void folderChecker(File folder) {
+        boolean result = folder.mkdir();
```

Comment:
You may want to use a name such as "isFolderCreated" that's more appropriate for boolean variables.